src/main/java/org/folio/circulation/resources/CheckOutByBarcodeResource.java

```diff
@@ -178,6 +197,39 @@ private CompletableFuture<Result<LoanAndRelatedRecords>> saveCheckOutSessionReco
         succeeded(records)));
   }
 
+  private CompletableFuture<Result<LoanAndRelatedRecords>> acquireLock(LoanAndRelatedRecords records, CheckOutLockRepository checkOutLockRepository, StringBuilder checkOutLockId) {
+    log.debug("acquireLock:: Creating checkout lock {} ", records.getCheckoutLockConfiguration());
+    if (records.getCheckoutLockConfiguration() == null || !records.getCheckoutLockConfiguration().isCheckOutLockFeatureEnabled()) {
+      return completedFuture(Result.succeeded(records));
+    }
+    CompletableFuture<CheckOutLock> future = new CompletableFuture<>();
+    checkOutLockRepository.createLockWithRetry(0, future, records);
+    return future.handle((res, err) -> {
+      if (err != null) {
+        log.error("acquireLock:: Unable to acquire lock for item {} ", records.getItem().getBarcode(), err);
+        return Result.failed(ValidationErrorFailure.singleValidationError("unable to acquire lock", "", ""));
+      }
+      checkOutLockId.append(res.getId());
+      return Result.succeeded(records);
+    });
+  }
+
+  private CompletableFuture<Result<LoanAndRelatedRecords>> validateItemLimitBasedOnLockFeatureFlag(LoanAndRelatedRecords records, CheckOutValidators validators, CirculationErrorHandler errorHandler) {
+    if (records.getCheckoutLockConfiguration() == null || !records.getCheckoutLockConfiguration().isCheckOutLockFeatureEnabled()) {
```

Comment:
Would be more DRY and more readable to put these two checks into a method that took care of this inside the `LoanAndRelatedRecords` class. You're doing the same check in `acquireLock`. This method could also be called `isCheckoutLockFeatureEnabled`.

Reply:
Created a new method in LoanAndRelatedRecords class and moved this logic.